25 user location tags in osm file

jasonrogena opened the pull request:
Closes #25

jasonrogena committed:
JR: Add User Location Tags in .osm File

Add logic for appending the user's location and location accuracy as
osm tags in the data returned to ODK Collect. Fixes issue #25

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

jasonrogena committed:
JR: Add Unit and Functional Tests for Issue #25 Fix

Add most of the unit and functional tests for the user location tags in
.osm file issue fix

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

jasonrogena committed:
Refine Location Dependent Tests

Refine test that require a GPS connection for code fixing issue #25.
Leads to much more consistent results

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

ukanga commented on app/src/main/java/org/redcross/openmapkit/tagswipe/TagSwipeActivity.java:
```diff
@@ -65,6 +100,225 @@ protected void onCreate(Bundle savedInstanceState) {
+            Log.w("GPSTest", "LocationManager is not null");
+            Location location = locationManager.getLastKnownLocation(preferredLocationProvider);
+            if(location == null) {
+                Log.w("GPSTest", "Last location is null");
```
This test logs, were they to be removed or they should still be present? If present, should the logging use the class name and class method be part of the log or is this automatically added?

jasonrogena replied:
@ukanga They were meant to be removed. Let me do that

jasonrogena commented:
@ukanga They were meant to be removed. Let me do that

jasonrogena committed:
JR: Remove GPS Test Debug Logging

Remove Logging lines used to test GPS connectivity in TagSwipeActivity.
Part of fix for issue #25

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

ukanga commented:
This needs to be rebased, has merge conflicts.

jasonrogena commented:
The branch to be merged has some dependence on 'testMode' in the main code--similar to pull request #46. Please don't merge until changes from pull request #46 are rebased into this branch

jasonrogena committed:
Test Mode in TagSwipeActivity

Remove dependence on the testMode variable in the TagSwipeActivity

Fix for issue #25

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

jasonrogena commented:
Failing TagSwipeActivity tests referenced in issue #50
